src/de.rs

```diff
@@ -240,7 +254,7 @@ impl<'de, R: Read<'de>> Deserializer<R> {
         self.fix_position(err)
     }
 
-    fn deserialize_number<V>(&mut self, visitor: V) -> Result<V::Value>
+    fn deserialize_number<V>(&mut self, visitor: V, prim: bool) -> Result<V::Value>
```

Comment:
I think this method is only called when deserializing a primitive so it shouldn't need the new prim parameter.

Reply:
I'll double-check, but I don't think that's true. (Well, it wasn't when I originally wrote the code; that may have changed.)

Reply:
That wasn't true before, but now that I've removed the block you suggested above in `deserialize_struct`, it should be fine to get rid of.